watchface

```diff
@@ -0,0 +1,406 @@
+                exit
+            fi
+            ;;
+        clone|test)
+            declare -a args=("--title" "Watchfaces" "--clear" "--menu" "Choose a watchface:" 25 78 15 "--" )
+            for wf in "${WATCHFACES[@]}" ; do
+                args+=("${wf}" "")
+            done
+            if ! watchface=$("${MENUPROGRAM}" "${args[@]}" 3>&1 1>&2 2>&3); then
+                exit
+            fi
+            ;;
+    esac
+    if ("${MENUPROGRAM}" --clear --title "Confirming" --yesno "${action} ${watchface}" 0 0); then
+        echo "OK!"
+        if [ "${action}" = "clone" ] ; then
+            if newface=$("${MENUPROGRAM}" \
+                --title "Cloning ${watchface}" --clear \
+                --inputbox "Enter name of new watchface:" \
+                8 50 "newface" \
+                3>&1 1>&2 2>&3)
+            then
+                "${action}"face "${watchface}" "${newface}"
+            else
+                echo "No name entered"
+            fi
+        elif [ "${action}" = "deploy" ] ; then
+            local first=true
+            for wf in ${watchface}; do
+                "${action}"face "${wf}" "${first}"
+                first=false
+            done
+            if [ "${REBOOT}" = "true" ] ; then
+                rebootWatch
+            fi
+        else
+            "${action}"face "${wf}"
+        fi
+    else
+        echo "Canceled"
+    fi
+}
+
+# These are the defaults for SSH access
+WATCHPORT=22
+WATCHADDR=192.168.2.15
+# These are the defaults for local QEMU target
+QEMUPORT=2222
+QEMUADDR=localhost
+# Assume no ADB unless told otherwise
+ADB=false
+# Assume no GUI unless told otherwise
+GUI=false
+# we're not done until we say we are
+DONE=false
+# No wallpaper unless asked
+WALLPAPER=""
+# Assume Dialog unless unavailable
+if hash dialog 2>/dev/null; then
+    MENUPROGRAM=dialog
+else
+    MENUPROGRAM=whiptail
+fi
+
+
+while [[ $# -gt 0 ]] ; do
+    case $1 in
+        -a|--adb)
+            ADB=true
+            shift
+            ;;
+        -b|--boot)
+            REBOOT=true
+            shift
+            ;;
+        -g|--gui)
+            GUI=true
+            shift
+            ;;
+        -q|--qemu)
+            WATCHPORT=${QEMUPORT}
+            WATCHADDR=${QEMUADDR}
+            shift
+            ;;
+        -p|--port)
+            WATCHPORT="$2"
+            shift
+            shift
+            ;;
+        -r|--remote)
+            WATCHADDR="$2"
+            shift
+            shift
+            ;;
+        -h|--help)
+            showHelp
+            exit 1
+            ;;
+        -w|--wall)
+            WALLPAPER="$2"
+            shift
+            shift
+            ;;
+        clone)
+            cloneface "$2" "$3"
+            shift
+            shift
+            shift
+            DONE=true
+            ;;
+        deploy)
+            deployface "$2" true
+            shift
+            shift
+            DONE=true
+            ;;
+        test)
+            testface "$2"
```

Comment:
This testface/deployface/cloneface naming makes me think of the "boaty mc boatface" meme. In the context of a script which is called watchface it's probably explicit enough already and we can just name them clone/deploy/test no ?

Reply:
It's because `test` is a bash keyword that I changed all three to end in `face` so they wouldn't clash and the reader would be alerted to the fact that they're related.  I'd be open to some other name, but this was my way of resolving that.

Reply:
Alright, I guess I would have used "watchface" instead of "face" and rather "Watchface" to keep casing consistent with the other questions but I don't want to torture you further and tbh I find it amusing enough that I don't even mind.